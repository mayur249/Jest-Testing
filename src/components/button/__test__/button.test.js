import React from "react";
import ReactDOM from "react-dom";
import Button from "../button";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
const ReactTestRenderer = require('react-test-renderer');

afterEach(cleanup);

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Button />, div);
});

it("renders button correctly", () => {
  const { getByTestId } = render(<Button label="Click Me" />);
  expect(getByTestId("button")).toHaveTextContent("Click Me");
});

it("renders button correctly", () => {
  const { getByTestId } = render(<Button label="Dont Click Me" />);
  expect(getByTestId("button")).toHaveTextContent("Dont Click Me");
});

it("it matches snapshot 1", () => {
  const tree = ReactTestRenderer.create(<Button label="Dont Click Me" />).toJSON();
  expect(tree).toMatchSnapshot()
});

it("it matches snapshot 2", () => {
    const tree = ReactTestRenderer.create(<Button label="Dont Click Me Broo" />).toJSON();
    expect(tree).toMatchSnapshot()
  });
