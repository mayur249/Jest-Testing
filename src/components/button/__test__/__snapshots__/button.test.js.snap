// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshot 1 1`] = `
<div
  className="button-style"
  data-testid="button"
>
  Hello 
  Dont Click Me
</div>
`;

exports[`it matches snapshot 2 1`] = `
<div
  className="button-style"
  data-testid="button"
>
  Hello 
  Dont Click Me Broo
</div>
`;
